.github/workflows/ci.yaml

```diff
@@ -51,11 +51,12 @@ jobs:
   fedora:
     runs-on: ubuntu-latest
     container:
-      image: fedora:latest
+      image: ${{ matrix.image }}
```

Comment:
I think you can get away with just doing the tag in the matrix and not the whole image name